src/processing/schema.ts

```diff
@@ -1,10 +1,18 @@
-export const validateApiCallParameters = (apiCallParameters: unknown): ApiCallParameters => {
-  // eslint-disable-next-line lodash/prefer-lodash-typecheck
-  if (typeof apiCallParameters !== 'object' || apiCallParameters === null) {
-    throw new TypeError('Invalid API call parameters');
-  }
+import { z } from 'zod';
 
-  return apiCallParameters as ApiCallParameters;
-};
+export const apiCallParametersSchema = z.record(z.any());
```

Comment:
Thinking out loud again, but wouldn't a more generic name be better/more reusable here? I sometimes use something like `interface AnyObject { [key: string]: any; }` for these kinds of things.

There isn't really anything specific making this "apiCallParameter" related - that can be indicated through argument names

Reply:
I think of it the other way around. It's not about finding a name for a particular type, but instead find a type for a particular name.

Reply:
Wouldn't that lead to lots of duplicated types over time? Plus in a lot of cases the types live in another file, while parameter names are always linked to the current scope. If there's a single generic type, you only need to see it once to understand it everywhere.

But anyway, I don't think it worth bikeshedding this 😄 I'm happy with the current approach

Reply:
> Wouldn't that lead to lots of duplicated types over time? Plus in a lot of cases the types live in another file, while parameter names are always linked to the current scope. If there's a single generic type, you only need to see it once to understand it everywhere.

We need to have a schema specific for `apiCallParameters` because it can be used by the consumers. To address the duplicated types issue we could do `apiCallParametersSchema = anyObjectSchema` if there is a need for that. 

Another disadvantage of using generic type for different "business values" is when one of them changes, but commons is really small for that to become an issue.

> But anyway, I don't think it worth bikeshedding this 😄 I'm happy with the current approach

I think we are not at a bike shedding point 😄. I find it a nice topic to think about.

Reply:
> We need to have a schema specific for apiCallParameters because it can be used by the consumers

Do we though? As long as the consumer can get a quick understanding of what the expected input shape should be, then I don't think it should matter.

My main issue is that it replicating this pattern in other places leads to lots of extra types that you often need to keep track of in your head. If you have an "AnyObject", you only need to understand it once and you can apply it everywhere (and it should be relatively obvious what "AnyObject" means). I still think it should be the job of the function name and argument names to communicate the business concepts.

> Another disadvantage of using generic type for different "business values" is when one of them changes

Do you mean like if I have code that looks like:

```ts
import { type ApiCallParameters } from '@api3/commons';
const params: ApiCallParameters = { a: 1, b: 2 };
```

And we (for example), change `ApiCallParameters` to be some other structure in a new version. The user would then only be required to update the value and leave the type unchanged? I personally don't really find this to be that big of a problem, but I can understand it might be for others.

Reply:
As I typed out the example above, I found myself opening up more to having the `ApiCallParameters` type haha. I think for a _library_ (this this project), it probably actually makes more sense to have the type as `ApiCallParameters`. For an _application_, I still kinda like `AnyObject`